blocks/algorithms/__init__.py

```diff
@@ -169,6 +170,10 @@ def __init__(self, step_rule=None, gradients=None, **kwargs):
                 zip(self.params, tensor.grad(self.cost, self.params))))
         self.step_rule = step_rule if step_rule else SteepestDescent()
 
+        self.total_gradient_norm = named_copy(
```

Comment:
Maybe on the long term it would be useful to store all these variables in an attribute `TrainingAlgorithm.monitoring_channels` (or whatever term we want to use). In that case a user can just go through the algorithms `monitoring_channels` attribute to see what the algorithm suggests in terms of monitoring values. That's easier than having to go through the documentation of each algorithm to see what monitors it makes public that you could monitor.